### PR TITLE
lammps: add user-omp package

### PR DESCRIPTION
**Pull request by junghans**
@louisvernon

### CODE REVIEW

**Comment by junghans** [comment]
@scheibelp

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def url_for_version(self, version):
     depends_on('mpi', when='+mpi')
     depends_on('fftw', when='+ksapce')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/lammps/package.py]: This was not introduced by your PR but would you mind correcting this to `+kspace`?

Comment (junghans) [reply var/spack/repos/builtin/packages/lammps/package.py]: Done!

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 2]
```diff
@@ -57,9 +57,9 @@ def url_for_version(self, version):
             description='Build with mpi')
 
     depends_on('mpi', when='+mpi')
-    depends_on('fftw', when='+ksapce')
+    depends_on('fftw', when='+kspace')
     depends_on('voropp', when='+voronoi')
-    depends_on('netcdf+mpi', when='+user-nc-dump')
+    depends_on('netcdf+mpi', when='+user-nefcdf')
```
[the remark this answers is not in the document]
Comment (junghans) [reply var/spack/repos/builtin/packages/lammps/package.py]: Fixed!

**Pull request merged**